nilearn/conftest.py

```diff
@@ -335,3 +328,111 @@ def img_atlas(shape_3d_default, affine_mni):
             "csf": 3,
         },
     }
+
+
+# ------------------------ SURFACE ------------------------#
+
+
+@pytest.fixture
+def mini_mesh():
+    """Small mesh for tests with 2 parts with different numbers of vertices."""
+    left_coords = np.asarray([[0.0, 0, 0], [1, 0, 0], [0, 1, 0], [0, 0, 1]])
+    left_faces = np.asarray([[1, 0, 2], [0, 1, 3], [0, 3, 2], [1, 2, 3]])
+    right_coords = (
+        np.asarray([[0.0, 0, 0], [1, 0, 0], [1, 1, 0], [0, 1, 0], [0, 0, 1]])
+        + 2.0
+    )
+    right_faces = np.asarray(
+        [
+            [0, 1, 4],
+            [0, 3, 1],
+            [1, 3, 2],
+            [1, 2, 4],
+            [2, 3, 4],
+            [0, 4, 3],
+        ]
+    )
+    return {
+        "left_hemisphere": InMemoryMesh(left_coords, left_faces),
+        "right_hemisphere": InMemoryMesh(right_coords, right_faces),
+    }
+
+
+@pytest.fixture
+def make_mini_img(mini_mesh):
+    """Small surface image for tests."""
+
+    def f(shape=()):
+        data = {}
+        for i, (key, val) in enumerate(mini_mesh.items()):
+            data_shape = tuple(shape) + (val.n_vertices,)
+            data_part = (
+                np.arange(np.prod(data_shape)).reshape(data_shape) + 1.0
+            ) * 10**i
+            data[key] = data_part
+        return SurfaceImage(mini_mesh, data)
+
+    return f
+
+
+@pytest.fixture
+def mini_mask(mini_img):
+    """Small surface mask."""
+    data = {k: (v > v.ravel()[0]) for k, v in mini_img.data.items()}
+    return SurfaceImage(mini_img.mesh, data)
+
+
+@pytest.fixture
+def mini_img(make_mini_img):
+    """Small surface image for tests."""
+    return make_mini_img()
+
+
+@pytest.fixture
+def flip():
+    """Flip first and second dictionary keys."""
```

Comment:
Not exactly it puts the last one at the first place. But I'm confused: dictionary keys are not sorted ?

Reply:
Yes this is not accuarate, I'll update the docstring. it's actually just taking the final key and putting it first. No they are not sorted but are ordered (can be sorted). but this is just for testing some checks looking for a mismatch between data and vertices shapes for a particular key